streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -235,10 +238,26 @@
             this.validTransitions.addAll(Arrays.asList(validTransitions));
         }
 
+        public boolean hasNotStarted() {
+            return equals(CREATED);
+        }
+
         public boolean isRunningOrRebalancing() {
             return equals(RUNNING) || equals(REBALANCING);
         }
 
+        public boolean isShuttingDown() {
+            return equals(PENDING_SHUTDOWN) || equals(PENDING_ERROR);
+        }
+
+        public boolean hasCompletedShutdown() {
+            return equals(NOT_RUNNING) || equals(ERROR);
+        }
+
+        public boolean hasStartedOrFinishedShuttingDown() {
```

Comment:
Technically this one is not used anywhere yet, but I intend to use it in [#10788](https://github.com/apache/kafka/pull/10788) so I just went ahead and added it here. Happy to remove it until it's actually needed if anyone prefers

Reply:
Maybe have it be `isShuttingDown() || hasStartedOrFinishedShuttingDown()` to be less brittle?